packages/settingeditor-extension/src/settingeditor.ts

```diff
@@ -604,64 +600,92 @@ namespace Private {
    * Create a plugin list item.
    */
   export
-  function createListItem(plugin: ISettingRegistry.IPlugin, annotations: ISettingRegistry.IPluginAnnotations): HTMLLIElement {
-    const annotation = annotations && annotations.annotation;
-    const caption = annotation && annotation.caption || plugin.id;
-    const className = annotation && annotation.className || '';
-    const iconClass = `${PLUGIN_ICON_CLASS} ${
-      annotation && annotation.iconClass || ''
-    }`;
-    const iconLabel = annotation && annotation.iconLabel || '';
-    const label = (annotation && annotation.label) || plugin.id;
+  function createListItem(registry: ISettingRegistry, plugin: ISettingRegistry.IPlugin): HTMLLIElement {
+    const icon = getHint(ICON_CLASS_KEY, registry, plugin);
+    const iconClass = `${PLUGIN_ICON_CLASS}${icon ? ' ' + icon : ''}`;
+    const iconLabel = getHint(ICON_LABEL_KEY, registry, plugin);
+    const title = plugin.schema.title || plugin.id;
+    const caption = `(${plugin.id}) ${plugin.schema.description}`;
 
     return VirtualDOM.realize(
-      h.li({ className, dataset: { id: plugin.id }, title: caption },
+      h.li({ dataset: { id: plugin.id }, title: caption },
         h.span({ className: iconClass, title: iconLabel }),
-        h.span(label))
+        h.span(title))
     ) as HTMLLIElement;
   }
 
   /**
-   * Populate the fieldset with a specific plugin's annotation.
+   * Check the plugin for a rendering hint's value.
+   *
+   * #### Notes
+   * The order of priority for overridden hints is as follows, from most
+   * important to least:
+   * 1. Data set by the end user in a settings file.
+   * 2. Data set by the plugin author as a schema default.
+   * 3. Data set by the plugin author as a top-level key of the schema.
+   */
+  function getHint(key: string, registry: ISettingRegistry, plugin: ISettingRegistry.IPlugin): string {
+    // First, give priorty to checking if the hint exists in the user data.
+    let hint = plugin.data.user[key];
```

Comment:
Should be `plugin.data.user && plugin.data.user[key];`, and `plugin.data.composite && plugin.data.composite[key];` below.